Missing prototype for vsprintf

The arduino compile complains about this:

https://github.com/mcci-catena/Catena-mcciadk/blob/88afd42c64ed695327c25007c9bf634cd01ed801/src/lib/mcciadklib_snprintf.c#L145

The errors:

```console
/home/travis/arduino_ide/libraries/Catena-mcciadk/src/lib/mcciadklib_snprintf.c: In function 'McciAdkLib_Vsnprintf':
/home/travis/arduino_ide/libraries/Catena-mcciadk/src/lib/mcciadklib_snprintf.c:145:2: warning: implicit declaration of function 'vsnprintf' [-Wimplicit-function-declaration]
  nSprintf = vsnprintf(
  ^
```

Since warnings are not errors, the compile doesn't fail. But this is against MCCI policy so we should fix this.

--- a/src/lib/mcciadklib_snprintf.c
+++ b/src/lib/mcciadklib_snprintf.c
@@ -32,6 +32,8 @@ Revision history:
 */
 
 #include "mcciadk_baselib.h"
+
+#include <stdio.h>
 
 /****************************************************************************\
 |

--- a/src/lib/mcciadklib_stringcomparecaseinsensitive.c
+++ b/src/lib/mcciadklib_stringcomparecaseinsensitive.c
@@ -72,6 +72,8 @@ McciAdkLib_StringCompareCaseInsensitive(
 
         for (;; ++pLeft, ++pRight)
                 {
+                int cLeft, cRight;
+
                 cLeft = *pLeft;
                 cRight = *pRight;
 

--- a/src/lib/mcciadklib_stringcomparecaseinsensitive.c
+++ b/src/lib/mcciadklib_stringcomparecaseinsensitive.c
@@ -64,8 +64,6 @@ McciAdkLib_StringCompareCaseInsensitive(
         const char *pRight
         )
         {
-        int cLeft, cRight;
-
         if (pLeft == NULL)
                 return pRight ? 0 : -1;
 
@@ -74,8 +72,6 @@ McciAdkLib_StringCompareCaseInsensitive(
 
         for (;; ++pLeft, ++pRight)
                 {
-                int cLeft, cRight;
-
                 cLeft = *pLeft;
                 cRight = *pRight;
 

--- a/src/mcciadk_baselib.h
+++ b/src/mcciadk_baselib.h
@@ -57,7 +57,7 @@ McciAdkLib_BufferToUlong(
 	bool *pfOverflow
 	);
 
-static size_t
+static inline size_t
 McciAdkLib_BufferToUint32(
 	const char *s,
 	size_t n,

